src/gc.c

```diff
@@ -4,35 +4,26 @@
 #include "config.h"
 #include "redismodule.h"
 #include "rmalloc.h"
+#include "module.h"
 #include <pthread.h>
 #include <assert.h>
+#include <unistd.h>
+#include "dep/thpool/thpool.h"
 
-static void BlockClients_push(BlockClients* ctx, RedisModuleBlockedClient* bClient) {
-  BlockClient* bc = rm_calloc(1, sizeof(BlockClient));
-  bc->bClient = bClient;
-  pthread_mutex_lock(&ctx->lock);
-  dllist_prepend(&ctx->clients, &bc->llnode);
-  pthread_mutex_unlock(&ctx->lock);
-}
-
-static RedisModuleBlockedClient* BlockClients_pop(BlockClients* ctx) {
-  pthread_mutex_lock(&ctx->lock);
-  RedisModuleBlockedClient* ret = NULL;
-  DLLIST_node* nn = dllist_pop_tail(&ctx->clients);
-  if (nn) {
-    BlockClient* bc = DLLIST_ITEM(nn, BlockClient, llnode);
-    ret = bc->bClient;
-    rm_free(bc);
-  }
-  pthread_mutex_unlock(&ctx->lock);
-  return ret;
+#define DEADBEEF (void*)0xBADC0FFEE0DDF00D
```

Comment:
Didn't we say we will do it without this DEADBEEF?

Reply:
Just trim for 32bit